client.go

```diff
@@ -866,12 +869,14 @@ type link struct {
 	err                error // err returned on Close()
 
 	// message receiving
-	paused        uint32        // atomically accessed; indicates that all link credits have been used by sender
-	receiverReady chan struct{} // receiver sends on this when mux is paused to indicate it can handle more messages
-	messages      chan Message  // used to send completed messages to receiver
-	buf           buffer        // buffered bytes for current message
-	more          bool          // if true, buf contains a partial message
-	msg           Message       // current message being decoded
+	paused              uint32              // atomically accessed; indicates that all link credits have been used by sender
+	receiverReady       chan struct{}       // receiver sends on this when mux is paused to indicate it can handle more messages
+	messages            chan Message        // used to send completed messages to receiver
+	pendingMessages     map[uint32]struct{} // used to send completed messages to receiver
+	pendingMessagesLock sync.RWMutex        // lock to protect concurrent access to pendingMessages
```

Comment:
pendingMessages map and associated lock on the link to track messages being handled